sdks/python/apache_beam/io/gcp/bigquery.py

```diff
@@ -496,6 +506,189 @@ def reader(self, test_bigquery_client=None):
         kms_key=self.kms_key)
 
 
+FieldSchema = collections.namedtuple('FieldSchema', 'fields mode name type')
+
+
+def _to_bool(value):
+  return value == 'true'
+
+
+def _to_decimal(value):
+  return decimal.Decimal(value)
+
+
+def _to_bytes(value):
+  """Converts value from str to bytes on Python 3.x. Does nothing on
+  Python 2.7."""
+  return value.encode('utf-8')
+
+
+class _JsonToDictCoder(coders.Coder):
+  """A coder for a JSON string to a Python dict."""
+
+  def __init__(self, table_schema):
+    self.fields = self._convert_to_tuple(table_schema.fields)
+    self._converters = {
+        'INTEGER': int,
+        'INT64': int,
+        'FLOAT': float,
+        'BOOLEAN': _to_bool,
+        'NUMERIC': _to_decimal,
+        'BYTES': _to_bytes,
+    }
+
+  @classmethod
+  def _convert_to_tuple(cls, table_field_schemas):
+    """Recursively converts the list of TableFieldSchema instances to the
+    list of tuples to prevent errors when pickling and unpickling
```

Comment:
This error was quite interesting. It seems that it's impossible to serialize and deserialize nested `bigquery.TableFieldSchema` instances:
```

from apache_beam.internal import pickler
from apache_beam.io.gcp.internal.clients import bigquery

obj = bigquery.TableFieldSchema(fields=[bigquery.TableFieldSchema()])
pickler.loads(pickler.dumps(obj))
```

This snippet triggers the following exception:
`AttributeError: 'FieldList' object has no attribute '_FieldList__field'`.

My workaround was to rewrite all TableFieldSchema instances to an equivalent tuple, which can be serialized and deserialized without problems.